packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts

```diff
@@ -161,24 +167,52 @@ export class StatefulDagLMDGHOST implements ILMDGHOST {
   /**
    * Last finalized block
    */
-  private finalized: Node|null;
+  private finalized: {node: Node; epoch: Epoch} | null;
 
   /**
    * Last justified block
    */
-  private justified: Node|null;
+  private justified: {node: Node; epoch: Epoch} | null;
+  /**
+   * Best justified checkpoint.
+   */
+  private bestJustifiedCheckpoint: Checkpoint;
   private synced: boolean;
 
-  public constructor() {
+  public constructor(config: IBeaconConfig) {
     this.aggregator =
       new AttestationAggregator((hex: string) => this.nodes[hex] ? this.nodes[hex].slot : null);
     this.nodes = {};
     this.finalized = null;
     this.justified = null;
     this.synced = true;
+    this.config = config;
   }
 
-  public addBlock(slot: Slot, blockRootBuf: Hash, parentRootBuf: Hash): void {
+  /**
+   * Start method, should not wait for it.
+   * @param genesisTime
+   */
+  public async start(genesisTime: number): Promise<void> {
+    this.genesisTime = genesisTime;
+    const numSlot = computeSlotsSinceEpochStart(this.config, getCurrentSlot(this.config, this.genesisTime));
+    const timeToWaitTillNextEpoch = (this.config.params.SLOTS_PER_EPOCH - numSlot) * 
+      this.config.params.SECONDS_PER_SLOT * 1000;
+    // Make sure we call onTick at start of each epoch
+    await sleep(timeToWaitTillNextEpoch);
+    const epochInterval = this.config.params.SLOTS_PER_EPOCH * this.config.params.SECONDS_PER_SLOT * 1000;
+    setInterval(this.onTick.bind(this), epochInterval);
```

Comment:
We should have a `stop` method that does `clearInterval` that we call in `chain.stop`